testsuite/tests/typing-local/local.ml

```diff
@@ -2253,3 +2253,246 @@ module F :
   functor (X : sig val foo : (float -> local_ string) bi end) ->
     sig val foo : (float -> string) bi end
 |}]
+
+
+(*
+ * constructor arguments global
+ *)
+
+(* Global argument are preserved in module inclusion *)
+(* TODO: what about non-local? *)
+module M : sig
+  type t = Bar of global_ string
+end = struct
+  type t = Bar of string
+end
+[%%expect{|
+Lines 3-5, characters 6-3:
+3 | ......struct
+4 |   type t = Bar of string
+5 | end
+Error: Signature mismatch:
+       Modules do not match:
+         sig type t = Bar of string end
+       is not included in
+         sig type t = Bar of global_ string end
+       Type declarations do not match:
+         type t = Bar of string
+       is not included in
+         type t = Bar of global_ string
+       Constructors do not match:
+         Bar of string
+       is not compatible with:
+         Bar of global_ string
+       The second is global and the first is not.
```

Comment:
This error message is a bit confusing. It says that the constructor is global where as it is the argument to the constructor that is global. This would be even more confusing with multiple arguments. Something along the lines of:
```
The second has a global argument and the first does not
```
would be good for the single argument case. Whilst something like:
```
The second has a global argument at position 3 and the first does not.
```
would be needed when there are more arguments.

Reply:
Yes thanks for bringing this up.